### ISSUE TITLE
Lint for integer conversions using 'as'

### ISSUE THREAD

**Issue by martinthomson** [rust-lang/rust-clippy#4376]
#3947 talks about lints for checked conversions with TryFrom/TryInto, but most of the problems with integers is completely unchecked conversion.  In the code I'm working in to given an example, there are lots of conversions between `usize` and `u64`.

For instance, this doesn't warn, but it overflows on some platforms:
```rust
let x: u64 = 12;
let y: usize = x as usize;
```

Having been maintaining C code with rampant use of this style of unchecked conversion, this stuff can really get away from you.  It might not be something that can be enabled by default, but it could prevent some real bugs for those who are able to enable it.

**Comment by mikerite** [comment]
IMO, there already lints for this. Just search for 'cast' on the [lint docs](https://rust-lang.github.io/rust-clippy/master/).

The one that should cover this is [cast_possible_truncation](https://rust-lang.github.io/rust-clippy/master/index.html#cast_possible_truncation) and [it does](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=937e506b979190492928d0e5c4003d01).

**Comment by martinthomson** [comment]
LGTM, thanks.
